packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -49,42 +50,26 @@ describe('ReactDOMInput', () => {
     var node = ReactDOM.findDOMNode(stub);
     expectDev(console.error.calls.count()).toBe(1);
 
-    // Simulate a native change event
-    setUntrackedValue(node, 'giraffe');
+    setUntrackedValue.call(node, 'giraffe');
 
     // This must use the native event dispatching. If we simulate, we will
     // bypass the lazy event attachment system so we won't actually test this.
-    var nativeEvent = document.createEvent('Event');
-    nativeEvent.initEvent('change', true, true);
-    node.dispatchEvent(nativeEvent);
+    dispatchEventOnNode(node, 'change');
 
     expect(node.value).toBe('lion');
 
     document.body.removeChild(container);
   });
 
   it('should control a value in reentrant events', () => {
-    // This must use the native event dispatching. If we simulate, we will
-    // bypass the lazy event attachment system so we won't actually test this.
-    var inputEvent = document.createEvent('Event');
-    inputEvent.initEvent('input', true, true);
-    // This must use the native event dispatching. If we simulate, we will
-    // bypass the lazy event attachment system so we won't actually test this.
-    var changeEvent = document.createEvent('Event');
-    changeEvent.initEvent('change', true, true);
-
     class ControlledInputs extends React.Component {
       state = {value: 'lion'};
       a = null;
       b = null;
       switchedFocus = false;
       change(newValue) {
         this.setState({value: newValue});
-        // Calling focus here will blur the text box which causes a native
-        // change event. Ideally we shouldn't have to fire this ourselves.
-        // I don't know how to simulate a change event on a text box.
-        this.a.dispatchEvent(changeEvent);
-        this.b.focus();
+        dispatchEventOnNode(this.a, 'change');
```

Comment:
This has lost the `focus()` call. It is important. I verified that removing the fix in https://github.com/facebook/react/commit/84b8bbdfc9d47c82f3facc4e37d58463b5b54078 (where this test was introduced) fails on master, but doesn't fail here anymore.

Reply:
@gaearon Wow, I really wasn't aware of that fix. Thank you! 😃